encoding/dabatch.go

```diff
@@ -0,0 +1,244 @@
+package encoding
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/crypto"
+	"github.com/scroll-tech/go-ethereum/crypto/kzg4844"
+)
+
+// DABatchBase contains common metadata for all versions of DABatch
+type DABatchBase struct {
+	Version                uint8
+	BatchIndex             uint64
+	L1MessagePopped        uint64
+	TotalL1MessagePopped   uint64
+	DataHash               common.Hash
+	ParentBatchHash        common.Hash
+	SkippedL1MessageBitmap []byte
+}
+
+// DABatchV0 contains metadata about a batch of DAChunks.
+type DABatchV0 struct {
+	DABatchBase
```

Comment:
Not sure if we need to indirection of `DABatchBase`. Why not just make it `DABatchV0` and embed that one into `DABatchV1`?

Reply:
removed in [2a63797](https://github.com/scroll-tech/da-codec/pull/25/commits/2a63797f7c00370e8f51fe3e7ca9a659a0e47390).